### PR TITLE
Upgrade to Angular 8.2.0

### PR DESCRIPTION
**Pull request by radu-matei**
This mega-PR-of-doom upgrades Kashti to use Angular 8.2.0.

As expected, the process was not quite straightforward, and combined with the fact that I started this branch 3 months ago, let's say I'm happy the app builds and starts 😂 

TODO:

- [x] go through commit history and ensure no recent changes were lost
- [x] fix failing tests
- [x] update prod and dev images to Node 12

### CODE REVIEW

**Comment by flynnduism** [comment]
Tested this locally and it runs nice and smooth!

**Comment by radu-matei** [comment]
There is one more test failing:

```
Chromium 76.0.3809 (Ubuntu 0.0.0) StyleGuideComponent should create FAILED                                                                                              ReferenceError: $ is not defined                                                                                                                                error properties: Object({ ngDebugContext: DebugContext_({ view: Object({ def: Object({ factory: Function, nodeFlags: 33669121, rootNodeFlags: 33554433, nodeMatchedQueries: 0, flags: 0, nodes: [ Object({ nodeIndex: 0, parent: null, renderParent: null, bindingIndex: 0, outputIndex: 0, checkIndex: 0, flags: 33554433, childFlags: 114688, directChildFlags: 114688, childMatchedQueries: 0, matchedQueries: Object({  }), matchedQueryIds: 0, references: Object({  }), ngContentIndex: null, childCount: 1, bindings: [  ], bindingFlags: 0, outputs: [  ], element: Object({ ns: '', name: 'app-style-guide', attrs: [  ], template: null, componentProvider: Object({ nodeIndex: 1, parent: <circular reference: Object>, renderParent: <circular reference: Object>, bindingIndex: 0, outputIndex: 0, checkIndex: 1, flags: 114688, childFlags: 0, directChildFlags: 0, childMatchedQueries: 0, matchedQueries: Object, matchedQueryIds: 0, references: Object, ngContentIndex: -1, childCount: 0, bindings: Array, bindingFlags: 0, outputs:  ...
            at <Jasmine>
            at StyleGuideComponent.ngOnInit (http://localhost:9876/_karma_webpack_/webpack:/src/app/style-guide/style-guide.component.ts:15:5)
            at checkAndUpdateDirectiveInline (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:31909:1)
            at checkAndUpdateNodeInline (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44366:1)
            at checkAndUpdateNode (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44305:1)
            at debugCheckAndUpdateNode (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45327:36)
            at debugCheckDirectivesFn (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45270:1)
            at Object.eval [as updateDirectives] (ng:///DynamicTestModule/StyleGuideComponent_Host.ngfactory.js:9:5)
            at Object.debugUpdateDirectives [as updateDirectives] (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45258:1)
            at checkAndUpdateView (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44270:1)
            at callWithDebugContext (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45631:1)
Chromium 76.0.3809 (Ubuntu 0.0.0): Executed 31 of 40 (1 FAILED) (skipped 4) (0 secs / 0.942 secs)                                                               Chromium 76.0.3809 (Ubuntu 0.0.0) StyleGuideComponent should create FAILED
        ReferenceError: $ is not defined
        error properties: Object({ ngDebugContext: DebugContext_({ view: Object({ def: Object({ factory: Function, nodeFlags: 33669121, rootNodeFlags: 33554433, nodeMatchedQueries: 0, flags: 0, nodes: [ Object({ nodeIndex: 0, parent: null, renderParent: null, bindingIndex: 0, outputIndex: 0, checkIndex: 0, flags: 33554433, childFlags: 114688, directChildFlags: 114688, childMatchedQueries: 0, matchedQueries: Object({  }), matchedQueryIds: 0, references: Object({  }), ngContentIndex: null, childCount: 1, bindings: [  ], bindingFlags: 0, outputs: [  ], element: Object({ ns: '', name: 'app-style-guide', attrs: [  ], template: null, componentProvider: Object({ nodeIndex: 1, parent: <circular reference: Object>, renderParent: <circular reference: Object>, bindingIndex: 0, outputIndex: 0, checkIndex: 1, flags: 114688, childFlags: 0, directChildFlags: 0, childMatchedQueries: 0, matchedQueries: Object, matchedQueryIds: 0, references: Object, ngContentIndex: -1, childCount: 0, bindings: Array, bindingFlags: 0, outputs:  ...
            at <Jasmine>
            at StyleGuideComponent.ngOnInit (http://localhost:9876/_karma_webpack_/webpack:/src/app/style-guide/style-guide.component.ts:15:5)
            at checkAndUpdateDirectiveInline (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:31909:1)
            at checkAndUpdateNodeInline (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44366:1)
            at checkAndUpdateNode (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44305:1)
            at debugCheckAndUpdateNode (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45327:36)
            at debugCheckDirectivesFn (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45270:1)
            at Object.eval [as updateDirectives] (ng:///DynamicTestModule/StyleGuideComponent_Host.ngfactory.js:9:5)
            at Object.debugUpdateDirectives [as updateDirectives] (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:45258:1)
            at checkAndUpdateView (http://localhost:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2015/core.js:44270:1)
Chromium 76.0.3809 (Ubuntu 0.0.0): Executed 36 of 40 (1 FAILED) (skipped 4) (1.333 secs / 1.067 secs)                                                           TOTAL: 1 FAILED, 35 SUCCESS
```

File: src/app/footer/footer.component.html [hunk 1]
```diff
@@ -5,8 +5,8 @@
         <li>
           <span>
             <a href="https://github.com/brigadecore/kashti" target="_blank">KASHTI</a> and
-            <a href="https://github.com/brigadecore/brigade" target="_blank">BRIGADE</a> are licensed under the The
-            Apache 2.0 License.</span>
+            <a href="https://github.com/brigadecore/brigade" target="_blank">BRIGADE</a> are licensed under the The MIT
```
Comment (vdice) [inline src/app/footer/footer.component.html]: This should actually stay `Apache-2.0`, right?

File: src/app/project/project.component.spec.ts [hunk 2]
```diff
@@ -86,10 +99,13 @@ describe('ProjectComponent', () => {
       expect(component).toBeTruthy();
     });
 
-    it('should create when no project data is available', () => {
-      expect(component.project).toBeUndefined();
-      expect(component.builds).toBeUndefined();
-    });
-  });
+    // TODO - Radu M
```
Comment (vdice) [inline src/app/project/project.component.spec.ts]: Shall we create an issue around this TODO for a follow-up (if still applicable)?  Or shall we remove?

Comment (radu-matei) [reply src/app/project/project.component.spec.ts]: Yeah, I'd like to tackle this in a follow-up PR.

File: src/app/footer/footer.component.html (hunk 1 above)
Comment (radu-matei) [reply src/app/footer/footer.component.html]: Oh, you're right - seems like I missed cherry-picking that commit.
Nice catch, thanks!

File: README.md [hunk 3]
```diff
@@ -1,34 +1,27 @@
...
-Kashti is a web dashboard for Brigade, helping you easily visualize and inspect your
-Brigade builds. Kashti gives you a deep view into your Brigade projects, scripts,
-and jobs.
+## Code scaffolding
 
-This repo has a self-hosted [demo](https://brigadecore.github.io/kashti/)
+Run `ng generate component component-name` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module`.
 
-## Installation and Configuration
+## Build
```
[the remark this answers is not in the document]
Comment (radu-matei) [reply README.md]: This is a good catch - reverted to the original readme.

**Pull request merged**